megatron/training.py

```diff
@@ -252,6 +254,19 @@ def setup_model_and_optimizer(model_provider_func):
     optimizer, param_groups = get_optimizer(model)
     lr_scheduler = get_learning_rate_scheduler(optimizer)
 
+    # Determine if deepspeed config is JSON or filepath.
+    # If JSON then directly load it
+    deepspeed_conf = None
+    if hasattr(args, 'deepspeed_config'):
+        deepspeed_json_conf = args.deepspeed_config
+        if len(deepspeed_json_conf) > 2 and deepspeed_json_conf[0] == "'" and deepspeed_json_conf[-1] == "'":
+            deepspeed_json_conf = deepspeed_json_conf[1:-1]  # Remove shell quotes
+        try:
+            deepspeed_conf = json.loads(deepspeed_json_conf)
+            args.deepspeed_config = None
+        except JSONDecodeError:
```

Comment:
Should we just except this here? What if we are trying to read a json, but there's a parsing error?

Reply:
I wanted to add this here so if the user used the "old way" of defining a DS conf by providing a file path to a JSON then it would still work